llvm/test/CodeGen/CHERI-Generic/Inputs/atomic-rmw-cap-ptr-cap-arg.ll

```diff
@@ -0,0 +1,142 @@
+; Check that we can generate sensible code for atomic operations using capability pointers on capabilities
```

Comment:
Please remove all the unnecessary stuff from these tests: the block doesn't need a label, you can use `%0` in place of `%tmp`, you can remove the preds comments, nothing needs to be dso_local and you don't need the Function Attrs comments.